src/main/scala/dsptools/numbers/chisel_types/DspComplexTypeClass.scala

```diff
@@ -12,30 +12,51 @@ class DspComplexRing[T <: Data:Ring] extends Ring[DspComplex[T]] with hasContext
   def plus(f: DspComplex[T], g: DspComplex[T]): DspComplex[T] = {
     DspComplex.wire(f.real + g.real, f.imag + g.imag)
   }
+  def plusContext(f: DspComplex[T], g: DspComplex[T]): DspComplex[T] = {
+    DspComplex.wire(f.real context_+ g.real, f.imag context_+ g.imag)
+  }
   def times(f: DspComplex[T], g: DspComplex[T]): DspComplex[T] = {
+    val c_p_d = g.real + g.imag
```

Comment:
This feels like unnecessary code duplication. I'm wondering if we should have a function `threeMulComplexMultiply` and `fourMulComplexMultiply` that also takes an addition and multiplication function as parameters.

Then the `times(f, g)` would call `threeMulComplexMultiply(f, g, plus, multiply)` and `timesContext` would look at the context and either call `threeMulComplexMultiply(f, g, contextPlus, contextMultiply)` or `fourMulComplexMultiply(f, g, contextPlus, contextMultiply)` (with appropriate extra shift registers). What do you think?